lib/CodeGen/CGCall.cpp

```diff
@@ -3905,7 +3909,8 @@ RValue CodeGenFunction::EmitCall(const CGFunctionInfo &CallInfo,
 
         // If the argument doesn't match, perform a bitcast or an addrspacecast
         // to coerce it.  This can happen due to trivial type mismatches.
-        if (FirstIRArg < IRFuncTy->getNumParams() &&
+        if (V->getType()->isPointerTy() &&
```

Comment:
@yxsamliu so this is to ensure we don't incorrectly emit addrspacecast for non-pointer types, correct?